### PR TITLE
New column module in continuum mechanics

### PR DESCRIPTION
**Pull request by lfverlaan**
New Column module in continuum mechanics

#### References to other Issues or PRs

"Fixes#28053"

#### Brief description of what is fixed or changed
A new module has been created named "column". This module aims to calculate reaction forces and displacements due to axial forces. The user can initialize a column with length, area and elastic modulus. Properties have been added so the user can adjust variables. Methods have been added where the user can apply supports and loads to the column. The module sets up the load equation using singularity functions.

#### Other comments
This is the first pull request for this module. The column module will be extended in future pull requests.

#### Release Notes

* physics.continuum_mechanics
   * Added a new `Column` class for modeling axially loaded structural elements, similar in structure to the `Beam` class. It supports applying axial point loads, fixed supports, and assembling the load equation using singularity functions.

### CODE REVIEW

**Comment by lfverlaan** [comment]
The tests have been committed. Now the functions will be written in the column.py file to make sure that the tests will run successfully.

File: sympy/physics/continuum_mechanics/tests/test_column.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    # Test the variable setter
+    c.variable = x_1
+    assert c.variable == x_1
+
+    # Test joining two columns
+    c1 = Column(20, E_1, A_1)
+
+    c_joined = c.join(c1)
+    assert c_joined.length == 40
```
Comment (Tom-van-Woudenberg) [inline sympy/physics/continuum_mechanics/tests/test_column.py:44]: Isn't this join thing a results of the old way of dealing with multiple parts? How is this altered by Macaulay's method in the beam implementation? @lfverlaan

**Comment by Tom-van-Woudenberg** [comment]
> The tests have been committed. Now the functions will be written in the column.py file to make sure that the tests will run successfully.
> 
> The idea is to start with two boundary conditions (axial forces are zero just outside of the ends of the column) when initializing the class. When a support gets applied, a boundary condition for the deflection gets added. Now only simple instances will be explored, a column with two fixed ends and a column with one fixed and one free end. Any feedback on this approach?

Does the beam module function in a similar way (assuming zero axial forces)?

Would the functionality change when supports are added allongside the beam? If no other functions other required, it might be nice to add a test on that as well, making clear the scope of your implementation.

**Comment by lfverlaan** [comment]
The beam module uses the conditions that the shear force and the moment just right of the end is equal to zero (with the limit functions). 

I would think that the functionality does not change if more supports are applied, only more deflection boundary conditions are then added. I am not sure that I understand your question perfectly, but do you mean that if there is no extra function needed for extra supports that these can directly tested? And that if columns with more than two supports can already be calculated that they should just directly be included in the scope?

File: sympy/physics/continuum_mechanics/tests/test_column.py (hunk 1 above)
Comment (lfverlaan) [reply sympy/physics/continuum_mechanics/tests/test_column.py:44]: I haven't looked much at how the beam module deals with different parts. I thought the function would be a good basis for if and when I would like to work with different areas and elastic module. I also felt that the join function could be a clear and user friendly manner to apply hinges with in the future. I guess that if this was an old method and there is a better way using macaulay's method I will have to research it further. Thanks!

**Comment by Tom-van-Woudenberg** [comment]
> but do you mean that if there is no extra function needed for extra supports that these can directly tested?

Yes

> And that if columns with more than two supports can already be calculated that they should just directly be included in the scope?

Yes, or at least note that effectively your code works for a bit more than just the scope

**Comment by lfverlaan** [comment]
A base for the Column class has been committed. The code has a structure very similar to the Beam module, as the necessary processes up to this point are largely the same. One major difference is that there is no distinction between types of supports; in the Column module, a support refers to any constraint that is fixed in the horizontal direction.

The main goals of this issue has been resolved: the load equation can now be set up for columns with multiple supports and point loads. An example use case of the module is shown below:

```
from sympy.physics.continuum_mechanics.column import Column
from sympy.core.symbol import (Symbol, symbols)

L, E, A, F = symbols('L E A F')
c = Column(L, E, A)

c.apply_support(0)
c.apply_support(L / 2)
c.apply_support(L)

c.apply_load(F, L / 4, -1)
c.apply_load(F, 3*L / 4, -1)

print(c.load)
```
> F*SingularityFunction(x, L/4, -1) + F*SingularityFunction(x, 3*L/4, -1) + R_0*SingularityFunction(x, 0, -1) + R_L*SingularityFunction(x, L, -1) + R_L/2*SingularityFunction(x, L/2, -1)

I would love hear any feedback!

**Comment by Tom-van-Woudenberg** [comment]
Seems okay to me! Maybe add release note input here: https://github.com/sympy/sympy/pull/28049#issue-3045958183
@saiudayagiri might want to have a look too

If it's ready for a review by SymPy Core developers, please remove the draft label and tag Jason Moore

**Comment by Tom-van-Woudenberg** [comment]
> ```
> from sympy.physics.continuum_mechanics.column import Column
> from sympy.core.symbol import (Symbol, symbols)
> 
> L, E, A, F = symbols('L E A F')
> c = Column(L, E, A)
> 
> c.apply_support(0)
> c.apply_support(L / 2)
> c.apply_support(L)
> 
> c.apply_load(F, L / 4, -1)
> c.apply_load(F, 3*L / 4, -1)
> 
> print(c.load)
> ```
> 
> > F_SingularityFunction(x, L/4, -1) + F_SingularityFunction(x, 3_L/4, -1) + R_0_SingularityFunction(x, 0, -1) + R_L_SingularityFunction(x, L, -1) + R_L/2_SingularityFunction(x, L/2, -1)
> 

Maybe you could add this example to your docstring?

**Comment by Tom-van-Woudenberg** [comment]
I proposed to add documentation here: https://github.com/lfverlaan/sympy/pull/1

**Comment by lfverlaan** [comment]
> I proposed to add documentation here: [lfverlaan#1](https://github.com/lfverlaan/sympy/pull/1)

Great! I will make sure to merge it into my branch

**Comment by lfverlaan** [comment]
The pull request is ready for review. 

@moorepants

**Comment by Tom-van-Woudenberg** [comment]
@lfverlaan , this fixes error in the docs: https://github.com/lfverlaan/sympy/pull/2

**Comment by Tom-van-Woudenberg** [comment]
@saiudayagiri, since this is ready for review now, would you like to have a look at it?

**Comment by saiudayagiri** [comment]
@Tom-van-Woudenberg, @lfverlaan, PR #28049 looks good so far! I’ve seen the join method in Beam but I’m not sure of its implementation details; I need to look into that.

**Comment by Tom-van-Woudenberg** [comment]
> PR #28049 looks good so far! I’ve seen the join method in Beam but I’m not sure of its implementation details; I need to look into that.

@saiudayagiri, would you be able to review this now that you've been introduced properly?

File: sympy/physics/continuum_mechanics/column.py [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+            return ValueError("There is already a support at this location.")
+        
+        self._applied_supports.append(loc)
+        reaction_load = Symbol('R_' + str(loc))
+        self._load += reaction_load * SingularityFunction(self.variable, loc, -1)
+        self._bc_deflection.append((loc, 0))
+
+        return reaction_load
+    
+    def apply_load(self, value, start, order):
```
Comment (saiudayagiri) [inline sympy/physics/continuum_mechanics/column.py:159]: I don’t see an end parameter in the apply_load method.which is necessary for distributed loads

**Comment by saiudayagiri** [comment]
> > PR #28049 looks good so far! I’ve seen the join method in Beam but I’m not sure of its implementation details; I need to look into that.
> 
> @saiudayagiri, would you be able to review this now that you've been introduced properly?

Yes @Tom-van-Woudenberg and @lfverlaan — I’ve reviewed the PR #28049 and it looks good. From what I understand, this is primarily being developed for Structure2D integration, so the implementation seems appropriate. I did notice that apply_load currently lacks an end parameter, but I see that’s addressed in the next PR. The removal of the .join() method also makes sense, as it isn't necessary.

File: sympy/physics/continuum_mechanics/column.py [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+    def __init__(self, length, elastic_modulus, area, variable=Symbol('x'), base_char='C'):
+        """
+        Initializes the Column class.
+
+        Parameters
+        ==========
+
+        length: Sympifyable
+            A Symbol or value representing thecColumn's length.
+
```
Comment (saiudayagiri) [inline sympy/physics/continuum_mechanics/column.py:44]: Solve this issue as this got reviewed and open

**Comment by Tom-van-Woudenberg** [comment]
Superseded by https://github.com/sympy/sympy/pull/28138. @lfverlaan , could you close this one?
